Source/Core/VideoBackends/D3D/GeometryShaderCache.cpp

```diff
@@ -228,7 +221,7 @@ bool GeometryShaderCache::SetShader(u32 primitive_type)
 	}
 
 	// Need to compile a new shader
-	ShaderCode code = GenerateGeometryShaderCode(primitive_type, API_D3D);
+	ShaderCode code = GenerateGeometryShaderCode(primitive_type, API_D3D, uid.GetUidData());
```

Comment:
Yeah, I'm pretty sure primitive_type is in the UID. The `GetUidData()` could easily moved inside but I kind of like it better out here.